homeassistant/components/deconz/number.py

```diff
@@ -92,28 +109,35 @@ class DeconzNumber(DeconzDevice, NumberEntity):
     """Representation of a deCONZ number entity."""
 
     TYPE = DOMAIN
-
-    def __init__(self, device, gateway, description):
+    _device: Presence
+
+    def __init__(
+        self,
+        device: Presence,
+        gateway: DeconzGateway,
+        description: DeconzNumberEntityDescription,
+    ) -> None:
         """Initialize deCONZ number entity."""
-        self.entity_description = description
+        self.entity_description: DeconzNumberEntityDescription = description
         super().__init__(device, gateway)
 
-        self._attr_name = f"{self._device.name} {description.suffix}"
+        self._attr_name = f"{device.name} {description.suffix}"
         self._attr_max_value = description.max_value
         self._attr_min_value = description.min_value
         self._attr_step = description.step
 
     @callback
     def async_update_callback(self) -> None:
         """Update the number value."""
+        assert isinstance(self.entity_description, DeconzNumberEntityDescription)
         keys = {self.entity_description.update_key, "reachable"}
         if self._device.changed_keys.intersection(keys):
             super().async_update_callback()
 
     @property
     def value(self) -> float:
         """Return the value of the sensor property."""
-        return getattr(self._device, self.entity_description.device_property)
+        return float(getattr(self._device, self.entity_description.device_property))
```

Comment:
Doesn't this return a float? Did you mean to cast it, instead of parsing it?

Reply:
Thanks!